### PR TITLE
*mozilla: Allow appending of Additional items

### LINKED ISSUE #8016: [Request] Firefox
Hello.

I have been having a issue with Firefox package for over a year now but thought it wasn't worth raising it. But lately the firefox package manager has added some persisting files so maybe it can be accomodated.

Currently I use this:
https://github.com/xiaoxiaoflood/firefox-scripts

It adds the possibility of some modding to firefox that is not otherwise possible. It is mostly installed in the user profile, but 2 files are installed into the core firefox installation path. Whenever firefox updates itself by its own means, it persist perfectly, but scoop updates just install a new firefox so I would have to re add those files everytime. I raised the concern to xiaoxiao too but wanted to see scoop perspective.

Xiao adds files to the main firefox path, but also to defaults... so for instance making defaults persist would be a part way solution.

Overall firefox in scoop behaves a bit weird since at the end of the day firefox auto updates so a work around (bandaid) solution for me was to just "hold" firefox in scoop forever.

Just a shout out to see if there is a proper way to work this out.

### PR DESCRIPTION
**Pull request by HUMORCE**
USAGE:
Create a directory named **addition** in app's persist directory (e.g. ~\scoop\persist\firefox\addition), and put files/configs/etc. into the directory. 

Items of **addition** directory will be copied to app root directory by install script every install/update.

---

Any items that overwrite the original files are **NOT RECOMMENDED**.

see also: https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig

Closes #8016

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by HUMORCE** [comment]
@Sei-lus try and feedback.

File: bucket/firefox-esr.json [hunk 1]
```diff
@@ -20,7 +20,13 @@
         }
     },
     "extract_dir": "core",
-    "post_install": "firefox-esr -CreateProfile \"Scoop-ESR $persist_dir\\profile\"",
+    "post_install": [
+        "firefox-esr -CreateProfile \"Scoop-ESR $persist_dir\\profile\"",
+        "$Addition = \"$persist_dir\\addition\"",
+        "if (Test-Path $Addition) {",
+        "  info 'Copying additional items...'",
+        "  Copy-Item -Path $Addition\\* -Destination $dir -Force -Recurse -ErrorAction SilentlyContinue }"
```
Comment (niheaven) [inline bucket/firefox-esr.json:28]: Some notes: Could you move `}` to a separate line?

**Comment by Sei-lus** [comment]
@HUMORCE will give feedback as soon as I can.

**Comment by Sei-lus** [comment]
Ok, as good a time as ever, was waiting for the pull request to be added. Testing now.

**Comment by Sei-lus** [comment]
The addition folder had to be manually added. I would recommend to add documentation by default at persist.

i.e: Persist for firefox generates distribution and profile folders. I would recommend putting a .txt file indicating the usage of "addition" and adding "addition" folder by default to hint the users. Will have to wait firefox to go to 98 to see if it is working but I set it up.

**Comment by HUMORCE** [comment]
> The addition folder had to be manually added. I would recommend to add documentation by default at persist.

people who want to modding these apps can find the sugar. 🥶

**Pull request merged**